components/web_discovery/browser/ecdh_aes.cc

```diff
@@ -0,0 +1,137 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifdef UNSAFE_BUFFERS_BUILD
+#pragma allow_unsafe_buffers
+#endif
+
+#include "brave/components/web_discovery/browser/ecdh_aes.h"
+
+#include <array>
+
+#include "base/base64.h"
+#include "base/logging.h"
+#include "base/ranges/algorithm.h"
+#include "base/strings/string_number_conversions.h"
+#include "crypto/random.h"
+#include "crypto/sha2.h"
+#include "third_party/boringssl/src/include/openssl/aead.h"
+#include "third_party/boringssl/src/include/openssl/ec.h"
+#include "third_party/boringssl/src/include/openssl/ec_key.h"
+#include "third_party/boringssl/src/include/openssl/ecdh.h"
+#include "third_party/boringssl/src/include/openssl/nid.h"
+
+namespace web_discovery {
+
+namespace {
+
+constexpr size_t kAesKeySize = 16;
+constexpr size_t kAesTagLength = 16;
+constexpr size_t kIvSize = 12;
+constexpr size_t kKeyMaterialSize = 32;
+// P-256 field size * 2 + type byte
+constexpr size_t kComponentOctSize = 32 * 2 + 1;
+// type byte + public component + initialization vector
+constexpr size_t kEncodedPubKeyAndIv = 1 + kComponentOctSize + kIvSize;
+constexpr uint8_t kP256TypeByte = 0xea;
+
+bssl::UniquePtr<EC_KEY> CreateECKey() {
+  return bssl::UniquePtr<EC_KEY>(
+      EC_KEY_new_by_curve_name(NID_X9_62_prime256v1));
+}
```

Comment:
It feels to me very arbitrary having this function here for something that only occurs once, specially as it is not doing anything that improves reading the code in general. I think when we create a separate function we must **state** something. In this particular case I felt this function merely made it harder for me to reason why the function was here in the first place.